sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeResult.cs

```diff
@@ -12,19 +12,14 @@ namespace Azure.Search.Documents.Models
     /// <summary> The result of testing an analyzer on text. </summary>
     public partial class AnalyzeResult
     {
-        /// <summary> Initializes a new instance of AnalyzeResult. </summary>
-        internal AnalyzeResult()
-        {
-        }
-
         /// <summary> Initializes a new instance of AnalyzeResult. </summary>
         /// <param name="tokens"> The list of tokens returned by the analyzer specified in the request. </param>
-        internal AnalyzeResult(IList<TokenInfo> tokens)
+        internal AnalyzeResult(IReadOnlyList<TokenInfo> tokens)
```

Comment:
Needs argument validation for required fields. I have a NullOrEmpty check that would work for strings and enumerables (technically different, but same name) or we should at least verify these are null; otherwise, they are not required.

Reply:
https://github.com/Azure/autorest.csharp/issues/567